Refactor on top of Nan's changes to remove syntax errors

jakebiesinger opened the pull request:
@Nan-Zhang @Elmira88 @anbangx @JavierJia 

These changes are on top of @Nan-Zhang's recent refactoring.  In one fell swoop, Nan:
- removed the edge readids
- added `thisReadSequence`, `mateReadSequence`

which effectively obliterates `SplitRepeat` and `Scaffolding`.  This branch is based on Nan's branch and is just the changes to remove syntax errors that were present in his branch.  It's likely that stuff is broken.  Further, with these changes there are several optimizations we can make such as using `Kmer`s (not `VKmer`s) as the `vertexId` and in all the `Node.edges`. I haven't made any of those changes here. Finally, there are places where variable names should be changed (e.g., `edgeMap` doesn't make sense anymore).  I made some changes along those lines where I saw them, but I've likely missed a few.

@anbangx can you look closely at the pregelix-specific code to see if I made any stupid logic errors?

@Elmira88 you should be able to use this branch directly.  I'd suggest you `git cherry-pick` only the exact commits you want applied.  For example:

``` bash
$ git checkout <your-current-ray-branch>
$ git log
# record all hashes from the commits you've made for Ray.
# don't include any duplicates and don't include the commit where you merged with Nan's previous code
$ git checkout wbiesing/ray-genomix-refactor  # get these changes
$ git checkout elmira88/implementing-ray  # start a new branch based on this one
# now we will select only the commits you made before and reapply them onto this branch
# for example, say you want to start at commit a65asd6 and include all commits up until f756a7c
# let's call those START_COMMIT and END_COMMIT:
$ git cherry-pick START_COMMIT^..END_COMMIT  # don't forget the ^
$ git log  # notice that everything from START_COMMIT to END_COMMIT has now been applied *on top* of my commits in this branch
# repeat for each range of commits you want to add
```

Let's get the Ray scaffolding in place as soon as we can :v:

ramangrover committed:
added job info

ramangrover committed:
checkpoint

ramangrover committed:
merge from master

ramangrover committed:
checkpoint

ramangrover committed:
temporarily altered defaults for testing

ramangrover committed:
Merge branch 'master' into raman/master_feeds

ramangrover committed:
merge from master

ramangrover committed:
Merge branch 'master' into raman/master_feeds

ramangrover committed:
changes to tracked locations assigned to an operator for each partition

ramangrover committed:
Merge branch 'master' into raman/master_feeds

sigmod committed:
add client counter collection

sigmod committed:
let system_load be average of a specified period

sigmod committed:
add I/O counter

sigmod committed:
fix for the case iostat is not installed

sigmod committed:
add IO counters in the client library

sigmod committed:
fix the linux io counter

sigmod committed:
fix linux io counter to return relative I/Os

sigmod committed:
1. add deployment retry 2. support plan switch

ramangrover committed:
Merge branch 'master' into raman/master_feeds

sigmod committed:
fix jobgen

sigmod committed:
let pregelix driver print perf. counter

sigmod committed:
fix driver logging

ramangrover committed:
removed unnecessary fields in JobInfo

ramangrover committed:
Merge branch 'master' into raman/master_feeds

sigmod committed:
add the dyanmic optimization skeleton

jakebiesinger committed:
increase hardcoded pregelix framesize (tmp fix)

ramangrover committed:
restoring default values for heartbeat_period and max_heartbeat_elapsed, temporarily changed for fault tolerance experiments

ramangrover committed:
reformatting

ramangrover committed:
reformatting

ramangrover committed:
formatting

jakebiesinger committed:
increase default FRAME_LIMIT

jakebiesinger committed:
BUILD-profile optimize: use static final EDGETYPE.values not values()

jakebiesinger committed:
BUILD-profile optimize: only compile regex once

jakebiesinger committed:
BUILD-profile optimize: copy reference rather than deep copy in build

ramangrover committed:
Merge branch 'raman/master_feeds' of https://code.google.com/p/hyracks into raman/master_feeds

ramangrover committed:
removed repetitive comment

sigmod committed:
dealing with empty counter

jakebiesinger committed:
reformat

jakebiesinger committed:
BUILD-profile optimize: remove getLengths() and return offsets from copy ops

jakebiesinger committed:
MERGE-profile optimize: in/out formats use tmp references

jakebiesinger committed:
MERGE-profile optimize: less reading from conf; use global debug vals

sigmod committed:
add APT for retrieving slave counters

sigmod committed:
Merge branch 'master' of https://code.google.com/p/hyracks

jakebiesinger committed:
MERGE-profile optimize: random uses hash, not many calls, to break correlation

jakebiesinger committed:
fix to make pregelix and hyracks NC/CC's read from worker.logging.properties

Nan-Zhang committed:
update KMP and change name to indexOf()

jakebiesinger committed:
factor out edgeMap code in favor of VKmerList; remove SplitRepeat

jakebiesinger committed:
Merge pull request #67 from uci-cbcl/wbiesing/increase-pregelix-hardcoded-framesize

increase hardcoded pregelix framesize (tmp fix)

jakebiesinger committed:
MERGE-profile optimize: don't send unnecessary edges in MERGE

anbangx committed:
Fix no testcase bug

jakebiesinger committed:
MERGE-profile optimize: only send non-overlapping parts of kmers

anbangx committed:
Add bin to .gitignore

sigmod committed:
merge master

Nan-Zhang committed:
add random generate String function

Nan-Zhang committed:
add longer read test for IndexOf()

Nan-Zhang committed:
modify based on Jianfeng’s comments

Nan-Zhang committed:
add comments for IndexOf()

Nan-Zhang committed:
Merge branch 'genomix/fullstack_genomix' into nanzhang/VKmer_IndexOf_KMP

# By Jake Biesinger
# Via Jake Biesinger
* genomix/fullstack_genomix:
  increase default FRAME_LIMIT
  increase hardcoded pregelix framesize (tmp fix)

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/checker/SymmetryCheckerVertex.java:
```diff
@@ -62,14 +62,20 @@ public void checkSymmetry(Iterator<SymmetryCheckerMessage> msgIterator) {
-            if (!edgeMapIsSame)
+            
+            boolean edgesAreSame = true;
+            for (VKmer kmer : incomingMsg.getEdges()) {
```
Here you need to compare vertex.edge and msg.edge one by one. Only checking if vertex.edge contains msg.edge is not enough. I would like to mark both vertices if there is any asymmetric. In this code, it is only marked one vertex.

jakebiesinger replied:
You mean iterate over both lists and check for containment of each element, right?

jakebiesinger replied:
Done.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/io/message/BubbleMergeMessage.java:
```diff
@@ -66,18 +66,23 @@ public void reset() {
     }
 
-    public EdgeMap getMinorToBubbleEdgeMap() {
+    public VKmerList getMinorToBubbleEdgeMap() {
```
Replace Map with List anywhere!

jakebiesinger replied:
you mean "everywhere"?

jakebiesinger replied:
Done.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/io/message/PathMergeMessage.java:
```diff
@@ -46,8 +46,8 @@ public void setInternalKmer(VKmer internalKmer) {
 
-    public EdgeMap getEdgeMap(EDGETYPE edgeType) {
-        return getNode().getEdgeMap(edgeType);
+    public VKmerList getEdgeMap(EDGETYPE edgeType) {
```
Map ---> List

jakebiesinger replied:
Done.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/bubblemerge/ComplexBubbleMergeVertex.java:
```diff
@@ -39,8 +39,8 @@
 
-    private EdgeMap incomingEdgeList = null;
-    private EdgeMap outgoingEdgeList = null;
+    private VKmerList incomingEdgeList = null;
```
Let's be consistent with the name for edges. Don't use edgeList some places and use edges some other places

jakebiesinger replied:
Done. Changed all `EdgeMap`s and `EdgeList`s to `edges`.  If you see any strays, let me know.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/bubblemerge/ComplexBubbleMergeVertex.java:
```diff
@@ -286,16 +284,12 @@ public void broadcaseKillselfAndNoticeToUpdateEdges(BubbleMergeMessage incomingM
-            vertex.getEdgeMap(neighborToMeDir).remove(incomingMsg.getSourceVertexId());
-        } else {
-            readIds = new ReadIdSet();
+        if (vertex.getEdges(neighborToMeDir).contains(incomingMsg.getSourceVertexId())) {
```
if !contains(), throw exception! Neighber doesn't have the edge to vertex who sends msg to it?

jakebiesinger replied:
Yeah, that's a good addition.  Added it.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/io/message/BubbleMergeMessage.java:
```diff
@@ -66,18 +66,23 @@ public void reset() {
+        return node.getEdges(getMinorToBubbleEdgetype().mirror());
     }
 
     public void addNewMajorToBubbleEdges(boolean sameOrientation, BubbleMergeMessage msg, VKmer topKmer) {
```
Remove this function. This is for update readIds in edge. It doesn't need in our new version.

jakebiesinger replied:
Done.

Nan-Zhang commented:
I want to change the name of getEdgeMap() to something else in my branch, @jakebiesinger @anbangx @JavierJia @Elmira88 what do you guys think about? what's new name is proper?

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/bubblemerge/SimpleBubbleMergeVertex.java:
```diff
@@ -147,15 +138,6 @@ public void processSimilarSet() {
-                    outgoingMsg.addNewMajorToBubbleEdges(sameOrientation, curMsg, topMsg.getSourceVertexId());
-                    sendMsg(curMsg.getMajorVertexId(), outgoingMsg);
-
                     // 4. send message to delete vertices -- for deletedSet
```
Change the comment to 2. ... Make it in order

jakebiesinger replied:
Done.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/pathmerge/BasicPathMergeVertex.java:
```diff
@@ -127,15 +127,14 @@ public void updateNeighbors() {
             outgoingMsg.setFlag(outFlag);
             for (EDGETYPE mergeEdge : mergeEdges) {
```
Could you help me change this mergeEdge/mergeEdges name to mergeEdgeEdgeType/mergeEdgeEdgeTypes. Maybe you can come up with good names.

jakebiesinger replied:
`mergeET`?

anbangx replied:
mergeET and mergeETs?

jakebiesinger replied:
yeah, could be.  Or `mergeEdgeType` and `mergeEdgeTypes` if we don't like abbreviations.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/pathmerge/P1ForPathMergeVertex.java:
```diff
@@ -213,11 +213,10 @@ public void receiveToNeighbor(Iterator<PathMergeMessage> msgIterator) {
-                value.getEdgeMap(aliveToMe).unionAdd(incomingMsg.getInternalKmer(), deletedReadIds);
+            if (value.getEdges(deleteToMe).contains(deletedKmer)) {
+                value.getEdges(deleteToMe).remove(deletedKmer);
+                if (!value.getEdges(aliveToMe).contains(incomingMsg.getInternalKmer()))
```
otherwise, throw exception

jakebiesinger replied:
Another good addition. Done.

anbangx commented on genomix/genomix-pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/scaffolding/BasicBFSTraverseVertex.java:
```diff
@@ -191,10 +191,10 @@ public void sendMsgToPathNodeToAddCommondReadId(HashMapWritable<LongWritable, Pa
-            VKmer preKmer = kmerList.getPosition(pathAndEdgeTypeList.size() - 2);
-
-            vertex.getEdgeMap(prevToMe.mirror()).get(preKmer).add(commonReadId.get()); // Option: put 2 for test
+//            EDGETYPE prevToMe = edgeTypeList.get(pathAndEdgeTypeList.size() - 2);
```
Why did you comment these few lines?

jakebiesinger replied:
All they're doing is adding the common readId to one of the edgeMaps.  I wasn't sure if I should throw this function away completely or just comment out the parts that make no sense.

jakebiesinger replied:
Actually, the whole algorithm doesn't make much sense after this...

anbangx commented:
@jakebiesinger  General comment: after this REFACTOR, we should never see EdgeMap, but I still see a lot of them. For VKmerList, why don't we change its name to EdgeList. It is only used to store edges. EdgeList is much more readable. Additionally, when we try to get edgeList, some classes use getEdges() but some use getEdgeList(). Can we make it consistent? 

Anyway, thanks. Good job.

jakebiesinger commented:
> @jakebiesinger General comment: after this REFACTOR, we should never see EdgeMap, but I still see a lot of them.

Yeah, I agree.

> For VKmerList, why don't we change its name to EdgeList. It is only used to store edges. EdgeList is much more readable.

I think we should actually step away from "List"-- it's really a set not a list.  I think the name `edges` is better than specifying the datatype in the name.

> Additionally, when we try to get edgeList, some classes use getEdges() but some use getEdgeList(). Can we make it consistent?

Yeah I'll go back and refactor again.

jakebiesinger committed:
check both msg and vertex in symmetry checker

jakebiesinger committed:
add some error checking; remove unused functions

jakebiesinger committed:
rename all `edgeMap`s and `edgeList`s to `edges`

jakebiesinger commented:
Requested refactors are in but there's still the question of what to do with the BFS code.  The whole algorithm is centered on just adding a readId.  It seems like we might want to have that code for some other purpose maybe?

jakebiesinger commented:
> I want to change the name of getEdgeMap() to something else in my branch, @jakebiesinger @anbangx @JavierJia @Elmira88 what do you guys think about? what's new name is proper?

Already done in this branch, @Nan-Zhang

jakebiesinger committed:
fix coverage bug form short kmers in MERGE messages

anbangx commented:
@jakebiesinger  I still didn't see your changes on KmerList. Change it to EdgeSet or EdgeList or Edges?

jakebiesinger commented:
> @jakebiesinger I still didn't see your changes on KmerList. Change it to EdgeSet or EdgeList or Edges?

Maybe I'm misunderstanding.  The datatype for the edges are currently `VKmerList`s.  You want to change that name as well?  I think it's fine as it is-- it's a very generic list of VKmers which we use as the container for our edges.  I removed all references in `Node` and elsewhere to `*Map` and `*List` since the type doesn't matter _to `Node`_ but I don't think that means we should change the class name from `VKmerList` to, say, `EdgeList`.  The datatype really is a generic VKmerList...

anbangx commented:
ok, let's keep VKmerList.

jakebiesinger committed:
add version of tipremove that doesn't require graph compression

Commit:
Merge pull request #70 from uci-cbcl/nanzhang/VKmer_IndexOf_KMP

update KMP and change name to indexOf()

anbangx committed:
Merge branch 'genomix/fullstack_genomix' into anbangx/fix-no-testcase

anbangx committed:
Merge pull request #72 from uci-cbcl/anbangx/fix-no-testcase

Anbangx/fix no testcase

jakebiesinger committed:
allow global aggregators to be specified in xml

explicitly setting the aggregator in the PregelixJob constructors would
override any values read in from the conf's resources.

Instead, this commit doesn't set the conf explicitly and instead
specifies an array of aggregator class names which will always be in
place when `getGlobalAggregatorClasses` is called.

jakebiesinger committed:
api for specifying update state for activate() and voteToHalt()

jakebiesinger committed:
fix NPE when no custom aggregator is set

jakebiesinger committed:
add an "iteration complete" hook for aggregation/reporting across iterations

This commit allows the user to specify a class which will be called upon
completion of each pregelix iteration.  This allows us to perform a user-
specified action between iterations.

As an example, a PerIterationGlobalAggregatesHook is provided which
saves the complete set of global aggregator states from every iteration,
allowing the user to observe aggregates from all iterations.

The default hook does nothing.

The hook instance is attached directly to the PregelixJob so that it can
be retrieved by the Driver's caller.

jakebiesinger committed:
stop tipremove search if max length is surpassed

jakebiesinger committed:
add support for Hadoop Counters via job.setCounterAggregatorClass

The PregelixJob.setCounterAggregatorClass sets up a (user-specified)
global aggregator and an iterationComplete hook to save Counter values.

The user-specified Counter-based aggregator (must extend
HadoopCountersAggregator) is saved to HDFS in each iteration and should
be restart/snapshot-aware.

The usage for setting up counters is to make a call to
job.setCounterAggregatorClass.  After job completion, the Counters may
be retrieved from HDFS using BspUtils.getCounters(job).

Note that there is currently only one spot for iterationComplete hooks
and this behavior occupies it.

jakebiesinger committed:
add new example for Counters usage

jakebiesinger committed:
simplify worker logging output to go Console only (-> */nc/logs/hostname.log)

jakebiesinger committed:
Merge branch 'yingyi/fullstack_fix' into genomix/fullstack_genomix

jakebiesinger committed:
use Counters as partial value to simplify HadoopCountersAggregator

jakebiesinger committed:
Merge branch 'yingyi/fullstack_fix' into wbiesing/cleanup-statistics-aggregators

jakebiesinger committed:
use pregelix CounterAggregator for vertex compute statistics

anbangx committed:
Add bubblemergewithsearch message

anbangx committed:
Add bubblemergewithsearch vertex value

anbangx committed:
Add message flag

anbangx committed:
Add bubblemergewithsearch operator

anbangx committed:
Add arrayOfEdgeTypes

anbangx committed:
Add process in source after collecting paths

anbangx committed:
Make different functions' names for different functionalities

anbangx committed:
Add testsuite and prepare to test bubblemergewithsearch

anbangx committed:
Working in progress

anbangx committed:
Add input/output format for bubblemergewithsearch

anbangx committed:
Add max-search-length for bubblemergewithsearch

anbangx committed:
Add test case for BMWS

anbangx committed:
Add new reasonable test case

anbangx committed:
Fix a bug in message communication

anbangx committed:
begin BFS bubble merge pass test

anbangx committed:
Using EdgeTypeList instead of ArrayList<EdgeType>

anbangx committed:
Bubble with BFS pass test

anbangx committed:
Need to use deep copy in destination vertex

anbangx committed:
Add expected file for bubblemergewithsearch

anbangx committed:
New BubbleMerge pass test

anbangx committed:
Clean up code

anbangx committed:
Add comments for bubblemergewithsearch

anbangx committed:
Add more explanation

anbangx committed:
Revert graphbuild testsuite

JavierJia committed:
Update README.md

update the Acknowledgement to use yourkit.

jakebiesinger committed:
rename previousNodes to visitedNodes

jakebiesinger committed:
fix bug where MIN_LENGTH_TO_KEEP isn't kept between iterations

jakebiesinger committed:
make TipSearch consistent with other msgs: use Integer, not int for length

jakebiesinger committed:
remove conflict in comments

jakebiesinger committed:
Merge pull request #75 from uci-cbcl/wbiesing/fullstack_fix-pregelix-api-modifications-change-states-iteration-reporter-counters

Latest merge with master plus pregelix API changes

jakebiesinger committed:
Merge pull request #74 from uci-cbcl/wbiesing/logging-tweaks

Wbiesing/logging tweaks

jakebiesinger committed:
Merge pull request #76 from uci-cbcl/wbiesing/cleanup-statistics-aggregators

cleanup statistics aggregators

jakebiesinger committed:
Merge pull request #79 from uci-cbcl/wbiesing/fullstack_fix-pregelix-api-modifications-change-states-iteration-reporter-counters

Wbiesing/fullstack fix pregelix api modifications change states iteration reporter counters

jakebiesinger committed:
fix bug in coverage; simplify merging with truncated kmers in messages

jakebiesinger committed:
Merge remote-tracking branch 'origin/genomix/fullstack_genomix' into wbiesing/optimizations-from-profiling-BUILD_HYRACKS

Conflicts:
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/io/VertexValueWritable.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/DeBruijnGraphCleanVertex.java

anbangx committed:
Some refactors

anbangx committed:
Reformat

anbangx committed:
Some tweaks

anbangx committed:
Merge branch 'genomix/fullstack_genomix' into anbangx/implement-bubble-merge-with-search

anbangx committed:
Change int to Integer in message

anbangx committed:
Add detection to intersection between different sources

anbangx committed:
Pass destFile instead of destDir

anbangx committed:
Rename incoming/outgoing to reverse/forward

jakebiesinger committed:
Merge pull request #68 from uci-cbcl/wbiesing/optimizations-from-profiling-BUILD_HYRACKS

optimizations from profiling build_hyracks

jakebiesinger committed:
rename TIP_WITH_SEARCH to TIP; old version to TIP_SINGLE_NODE

jakebiesinger committed:
Merge pull request #73 from uci-cbcl/wbiesing/implement-tip-remove-with-search

add version of tipremove that doesn't require graph compression

anbangx committed:
update internalKmer up to MAX_BFS_LENGTH

anbangx committed:
Add comments for self- and extended-loop

anbangx committed:
Merge branch 'genomix/fullstack_genomix' into anbangx/implement-bubble-merge-with-search

Conflicts:
	genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java
	genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/EDGETYPE.java

anbangx committed:
Simplify preToMe.mirror().dir().mirror().edgeTypes()

anbangx committed:
Some comments

anbangx committed:
Store string as command line option

anbangx committed:
Deal with pruning last node's edges case

jakebiesinger committed:
Merge remote-tracking branch 'origin/nanzhang/ray-genomix' into wbiesing/ray-genomix-refactor

Conflicts:
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/pathmerge/BasicPathMergeVertex.java

jakebiesinger committed:
Merge remote-tracking branch 'origin/genomix/fullstack_genomix' into wbiesing/ray-genomix-refactor

Conflicts:
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/minicluster/GenerateGraphViz.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/type/EDGETYPE.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/type/EdgeMap.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/type/Node.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/type/ReadHeadSet.java
	genomix/genomix-
data/src/main/java/edu/uci/ics/genomix/type/VKmer.java
	genomix/genomix-
data/src/test/java/edu/uci/ics/genomix/type/EdgeMapTest.java
	genomix/genomix-
data/src/test/java/edu/uci/ics/genomix/type/NodeTest.java
	genomix/genomix-
data/src/test/java/edu/uci/ics/genomix/type/ReadIdSetTest.java
	genomix/genomix-
data/src/test/java/edu/uci/ics/genomix/type/VKmerTest.java
	genomix/genomix-
hadoop/src/main/java/edu/uci/ics/genomix/hadoop/contrailgraphbuilding/GenomixReducer.java
	genomix/genomix-
hadoop/src/main/java/edu/uci/ics/genomix/hadoop/graph/GraphStatistics.java
	genomix/genomix-
hyracks/src/main/java/edu/uci/ics/genomix/hyracks/graph/dataflow/AggregateKmerAggregateFactory.java
	genomix/genomix-
hyracks/src/main/java/edu/uci/ics/genomix/hyracks/graph/dataflow/ReadsKeyValueParserFactory.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/DeBruijnGraphCleanVertex.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/bubblemerge/SimpleBubbleMergeVertex.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/pathmerge/BasicPathMergeVertex.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/splitrepeat/SplitRepeatVertex.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/operator/unrolltandemrepeat/UnrollTandemRepeat.java
	genomix/genomix-
pregelix/src/main/java/edu/uci/ics/genomix/pregelix/testhelper/BubbleAddVertex.java

anbangx committed:
Reformat

anbangx committed:
Merge pull request #77 from uci-cbcl/anbangx/implement-bubble-merge-with-search

Anbangx/implement bubble merge with search

jakebiesinger committed:
fix to support non-default pregelix cc http ports

jakebiesinger committed:
Merge branch 'genomix/fullstack_genomix' into wbiesing/ray-genomix-refactor

Conflicts:
	genomix/genomix-
pregelix/src/test/java/edu/uci/ics/genomix/pregelix/jobgen/JobGenerator.java

jakebiesinger commented:
Okay, merging it in!
